java/vector/src/main/codegen/templates/AbstractPromotableFieldWriter.java

```diff
@@ -42,7 +42,7 @@
    * @param type
    * @return
    */
-  abstract protected FieldWriter getWriter(MinorType type);
+  abstract protected FieldWriter getWriter(ArrowType type);
```

Comment:
2 things:
 - this kind of change should be in its own pull request.
 - For performance reason we may not want to do it.